tutorials/misc/plot_ecog.py

```diff
@@ -11,192 +11,343 @@
+                     task='ictal', datatype='ieeg', extension='vhdr')
 
-###############################################################################
-# Now we get the :term:`trans` that transforms from our MRI coordinate system
-# to the head coordinate frame. This transform will be applied to the
-# data when applying the montage so that standard plotting functions like
-# :func:`mne.viz.plot_evoked_topomap` will be aligned properly.
+# then we'll use it to load in the sample dataset
+# Here we use a format (iEEG) that is only available in MNE-BIDS 0.7+, so it
+# will emit a warning on versions <= 0.6
+raw = read_raw_bids(bids_path=bids_path, verbose=False)
 
-trans = mne.channels.compute_native_head_t(montage)
-print(trans)
+# Pick only the ECoG channels, removing the EKG channels
+raw.pick_types(ecog=True)
 
-###############################################################################
-# Now that we have our montage, we can load in our corresponding
-# time-series data and set the montage to the raw data.
+# Load the data
+raw.load_data()
 
-# first we'll load in the sample dataset
-raw = mne.io.read_raw_edf(misc_path + '/ecog/sample_ecog.edf')
+# Then we remove line frequency interference
+raw.notch_filter([60], trans_bandwidth=3)
 
 # drop bad channels
-raw.info['bads'].extend([ch for ch in raw.ch_names if ch not in ch_names])
-raw.load_data()
 raw.drop_channels(raw.info['bads'])
-raw.crop(0, 2)  # just process 2 sec of data for speed
 
-# attach montage
-raw.set_montage(montage)
+# the coordinate frame of the montage
+print(raw.get_montage().get_positions()['coord_frame'])
+
+# Find the annotated events
+events, event_id = mne.events_from_annotations(raw)
+
+# Make a 25 second epoch that spans before and after the seizure onset
+epoch_length = 25  # seconds
+epochs = mne.Epochs(raw, events, event_id=event_id['onset'],
+                    tmin=13, tmax=13 + epoch_length, baseline=None)
+
+# And then load data and downsample.
+# .. note: This is just to save execution time in this example, you should
+#          not need to do this in general!
+epochs.load_data()
+epochs.resample(200)  # Hz, will also load the data for us
 
-# set channel types to ECoG (instead of EEG)
-raw.set_channel_types({ch_name: 'ecog' for ch_name in raw.ch_names})
 
 ###############################################################################
-# We can then plot the locations of our electrodes on our subject's brain.
-# We'll use :func:`~mne.viz.snapshot_brain_montage` to save the plot as image
-# data (along with xy positions of each electrode in the image), so that later
-# we can plot frequency band power on top of it.
+# Explore the electrodes on a template brain
+# ------------------------------------------
 #
-# .. note:: These are not real electrodes for this subject, so they
-#           do not align to the cortical surface perfectly.
+# Our electrodes are shown in ``mni_tal`` space, which corresponds to
```

Comment:
Maybe this isn't the best place to go into how to do the CVS morph to transform to `fsaverage` space so this could just be left out. Seems more like a developer detail.